inc/graphql.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * This file contains functionality needed to integrate with WPGraphQL.
+ *
+ * @package Byline_Manager
+ */
+
+namespace Byline_Manager;
+
+use WPGraphQL;
+
+add_action( 'graphql_register_types', __NAMESPACE__ . '\register_byline_types' );
+
+/**
+ * Register data types and field for supporting bylines on posts.
+ */
+function register_byline_types() {
+	// Register a text profile.
+	register_graphql_object_type(
+		'TextProfile',
+		[
+			'description' => __( 'A TextProfile type', 'byline-manager' ),
+			'fields'      => [
+				'text' => [
+					'type'        => 'String',
+					'description' => __( 'Text for a text profile', 'byline-manager' ),
+				],
+			],
+		]
+	);
+
+	// Register a union for both Profile and TextProfile types.
+	register_graphql_union_type(
+		'ProfileTypes',
+		[
+			'typeNames'   => [ 'Profile', 'TextProfile' ],
+			'resolveType' => function ( $profile ) {
+				if (
+					$profile instanceof WPGraphQL\Model\Post
+					&& PROFILE_POST_TYPE === $profile->post_type
+				) {
+					return 'Profile';
+				}
+
+				return 'TextProfile';
+			},
+		]
+	);
+
+	// Register options for byline format.
+	register_graphql_enum_type(
+		'BylineTextEnum',
+		[
+			'description' => __( 'Byline format', 'byline-manager' ),
+			'values'      => [
+				'TEXT'       => [
+					'value' => 'text',
+				],
+				'LINKS'      => [
+					'value' => 'links',
+				],
+				'POST_LINKS' => [
+					'value' => 'post_links',
+				],
+			],
+		]
+	);
+
+	register_graphql_object_type(
+		'Byline',
+		[
+			'description' => __( 'Byline manager byline', 'byline-manager' ),
+			'fields'      => [
+				'bylineText' => [
+					'args'        => [
+						'format' => [
+							'type'        => 'BylineTextEnum',
+							'description' => __( 'Choose the byline format.', 'byline-manager' ),
+						],
+					],
+					'type'        => 'String',
+					'description' => __( 'Byline text', 'byline-manager' ),
+					'resolve'     => function( WPGraphQL\Model\Post $post, array $args ) {
+						switch ( $args['format'] ) {
+							case 'links':
+								$byline = get_the_byline_links( $post->ID );
+								break;
+
+							case 'post_links':
+								$byline = get_the_byline_posts_links( $post->ID );
+								break;
+
+							case 'text':
+							default:
+								$byline = get_the_byline( $post->ID );
+								break;
+						}
+
+						return $byline;
```

Comment:
```suggestion
						switch ( $args['format'] ) {
							case 'links':
								return get_the_byline_links( $post->ID );

							case 'post_links':
								return get_the_byline_posts_links( $post->ID );
						}

						return get_the_byline( $post->ID );
```

Reply:
I debated this and decided to go for the more verbose version to make clear what the formats were. Can probably handle that in an inline comment just as well.